src/device/device_ledger.cpp

```diff
@@ -1663,6 +1719,13 @@ namespace hw {
         recv_len -= 32;
       }
 
+      if (use_view_tags)
+      {
+        ASSERT_X(recv_len>=1, "Not enough data from device");
+        memmove(&view_tag.data, &this->buffer_recv[offset], 1);
+        recv_len -= 1;
```

Comment:
I'd add ++ after offset here, in case stuff gets added later after this and goes out of sync